src/main/java/seedu/address/logic/commands/tag/EditCommand.java

```diff
@@ -0,0 +1,155 @@
+package seedu.address.logic.commands.tag;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.util.List;
+import java.util.Optional;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Edits a tag based on its displayed index in the tag list and updates all
+ * items associated with this tag.
+ */
+public class EditCommand extends Command {
+    public static final String COMMAND_WORD = "tag edit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Edits the tag identified by the index number used in the displayed tag list. "
+            + "Note that all associations with this tag will be updated.\n"
```

Comment:
Hmm, is `associations` a bit hard to understand here? Would it be better to be more verbose here and say something like `Note that all items that have been previously associated with this tag will be updated automatically to be associated with the updated tag.` as per the user guide? 

P.S. May also need to update the user guide: it currently says `contacts` when it's supposed to be generic right?